components/datetime/src/skeleton/helpers.rs

```diff
@@ -378,15 +410,31 @@ pub fn get_best_available_format_pattern<'data>(
                             continue;
                         }
                         Ordering::Greater => {
-                            // The requested field symbol is missing from the skeleton.
-                            distance += REQUESTED_SYMBOL_MISSING;
-                            missing_fields += 1;
-                            requested_fields.next();
-                            continue;
+                            // https://unicode.org/reports/tr35/tr35-dates.html#Matching_Skeletons
+                            // A requested skeleton that includes both seconds and fractional seconds (e.g. “mmssSSS”) is allowed
+                            // to match a dateFormatItem skeleton that includes seconds but not fractional seconds (e.g. “ms”).
+                            if !(matched_seconds
+                                && requested_field.symbol
+                                    == FieldSymbol::Second(fields::Second::FractionalSecond))
+                            {
+                                // The requested field symbol is missing from the skeleton.
+                                distance += REQUESTED_SYMBOL_MISSING;
+                                missing_fields += 1;
+                                requested_fields.next();
+                                continue;
+                            }
                         }
                         _ => (),
                     }
 
+                    if requested_field.symbol
+                        == FieldSymbol::Second(fields::Second::FractionalSecond)
+                        && skeleton_field.symbol
+                            == FieldSymbol::Second(fields::Second::FractionalSecond)
+                    {
+                        matched_seconds = true;
```

Comment:
Question: Could this retain the fractional second index and save a second loop through requested fields? See my `Thought:` above.

I'm listing this as a question, since it could be used in an optional perf follow-up, as listed above.